src/safeInnerHtml.tsx

```diff
@@ -114,49 +177,59 @@ class SafeInnerHtml extends Component {
     }
   }
 
-  chooseNode({ node, key }) {
+  chooseNode({ node, key }: NodeChoice): NodeExtended {
     if (node instanceof HTMLScriptElement) {
-      this.scripts.push({ key, code: node.textContent });
-      const span = node.ownerDocument.createElement("span");
+      this.scripts.push({ key, code: (node as HTMLScriptElement).textContent });
+      const span = (node as HTMLScriptElement).ownerDocument.createElement(
+        "span"
+      );
       span.setAttribute("style", "display: none;");
-      return span;
+      return span as unknown as NodeExtended;
     }
 
-    return node;
+    return node as unknown as NodeExtended;
   }
 
   // eslint-disable-next-line class-methods-use-this
-  selector(key, elementName = "") {
+  selector(key: Key, elementName = "") {
     return `${elementName}.${key}`;
   }
 
-  initialize({ children, decode = false }) {
-    this.innerHTML = unwrap(children);
+  initialize({ children, decode = false }: SafeInnerHtmlProps) {
+    this.innerHTML = unwrap(children as string[] | string);
     this.fragment = this.createFragment(this.innerHTML, decode);
     this.scripts = [];
     this.css = [];
   }
 
   // eslint-disable-next-line class-methods-use-this
-  createFragment(innerHTML, decode) {
+  createFragment(innerHTML: string, decode: boolean): ChildNode[] {
     const html = decode ? unescape(innerHTML) : innerHTML;
     return parseHTML(html);
   }
 
-  createElement({ localName, attributes, parentNode, childNodes, key }) {
-    const localNames = flow(map("localName"), compact)(attributes);
+  createElement({
+    localName,
+    attributes,
+    parentNode,
+    childNodes,
+    key,
+  }: ElementExtended): ReactElement | undefined {
+    const localNames = flow(map("localName"), compact)(attributes) as string[];
 
     const props = htmlProps(
       flow(
-        map((attribute) =>
-          this.chooseAttribute({ attribute, key, elementName: localName })
+        map((attribute: AttributeExtended) =>
+          this.chooseAttribute({ attribute, key, elementName: localName! })
```

Comment:
De interface van deze methode accepteert `undefined` voor `localName`, maar hier wordt - zonder dat duidelijk is waarom - aangenomen dat het _niet_ `undefined` is.